crates/fluvio/src/producer/mod.rs

```diff
@@ -90,24 +94,72 @@ impl ProducerPool {
         }
     }
 
-    fn shared(
+    #[allow(clippy::too_many_arguments)]
+    pub async fn add_producer_if_needed(
+        &mut self,
         config: Arc<TopicProducerConfig>,
         topic: String,
         spu_pool: Arc<SpuPool>,
-        batches: Arc<Vec<BatchHandler>>,
         client_metric: Arc<ClientMetrics>,
-    ) -> Arc<Self> {
-        Arc::new(ProducerPool::new(
+        partition_id: PartitionId,
+        batchs: Arc<BatchesDeque>,
+        batch_events: Arc<BatchEvents>,
+        record_accumulator: Arc<RecordAccumulator>,
+    ) -> Option<(Arc<BatchEvents>, Arc<BatchesDeque>)> {
+        if self.flush_events.contains_key(&partition_id) {
+            return None;
+        }
+
+        record_accumulator
+            .add_partition(partition_id, (batch_events.clone(), batchs.clone()))
+            .await;
+
+        self.add_producer(
             config,
             topic,
             spu_pool,
-            batches,
             client_metric,
-        ))
+            partition_id,
+            batchs.clone(),
+            batch_events.clone(),
+        );
+        Some((batch_events.clone(), batchs.clone()))
+    }
+
+    #[allow(clippy::too_many_arguments)]
+    pub fn add_producer(
```

Comment:
probably not made public.  seems to share same arg as other

Reply:
Yeah, should be private! Removed!